crates/parry3d/examples/custom_dispatcher.rs

```diff
@@ -0,0 +1,167 @@
+extern crate nalgebra as na;
+
+use parry3d::{
+    bounding_volume::{Aabb, BoundingSphere},
+    mass_properties::MassProperties,
+    math::{Isometry, Point, Vector},
+    query::{
+        ClosestPoints, Contact, DefaultQueryDispatcher, NonlinearRigidMotion, PointProjection,
+        PointQuery, QueryDispatcher, Ray, RayCast, RayIntersection, ShapeCastHit, ShapeCastOptions,
+        Unsupported,
+    },
+    shape::{Ball, Cuboid, FeatureId, Shape, ShapeType, TypedShape},
+};
+
+fn main() {
+    let cube = Cuboid::new(Vector::new(1.0, 1.0, 1.0));
+    let ball = CustomBall(Ball::new(1.0));
+
+    let pos12 = Isometry::identity();
+    let dispatcher = CustomBallDispatcher;
+
+    let contact = dispatcher.contact(&pos12, &cube, &ball.0, 0.0);
+
+    dbg!(contact);
+}
+
+pub struct CustomBall(pub Ball);
+
+impl PointQuery for CustomBall {
+    fn project_local_point(&self, pt: &Point<f32>, solid: bool) -> PointProjection {
+        self.0.project_local_point(pt, solid)
+    }
+
+    fn project_local_point_and_get_feature(&self, pt: &Point<f32>) -> (PointProjection, FeatureId) {
+        self.0.project_local_point_and_get_feature(pt)
+    }
+}
+
+impl RayCast for CustomBall {
+    fn cast_local_ray_and_get_normal(
+        &self,
+        ray: &Ray,
+        max_time_of_impact: f32,
+        solid: bool,
+    ) -> Option<RayIntersection> {
+        self.0
+            .cast_local_ray_and_get_normal(ray, max_time_of_impact, solid)
+    }
+}
+
+impl Shape for CustomBall {
+    fn compute_local_aabb(&self) -> Aabb {
+        self.0.compute_local_aabb()
+    }
+
+    fn compute_local_bounding_sphere(&self) -> BoundingSphere {
+        self.0.compute_local_bounding_sphere()
+    }
+
+    fn clone_dyn(&self) -> Box<dyn Shape> {
+        Box::new(Self(self.0))
+    }
+
+    fn scale_dyn(&self, scale: &Vector<f32>, num_subdivisions: u32) -> Option<Box<dyn Shape>> {
+        Some(self.0.scale_dyn(scale, num_subdivisions)?)
+    }
+
+    fn mass_properties(&self, density: f32) -> MassProperties {
+        self.0.mass_properties(density)
+    }
+
+    fn shape_type(&self) -> ShapeType {
+        self.0.shape_type()
+    }
+
+    fn as_typed_shape(&self) -> TypedShape {
+        self.0.as_typed_shape()
+    }
+
+    fn ccd_thickness(&self) -> f32 {
+        self.0.ccd_thickness()
+    }
+
+    fn ccd_angular_thickness(&self) -> f32 {
+        self.0.ccd_angular_thickness()
+    }
+}
+
+pub struct CustomBallDispatcher;
+
+impl QueryDispatcher for CustomBallDispatcher {
+    fn intersection_test(
+        &self,
+        pos12: &Isometry<f32>,
+        g1: &dyn Shape,
+        g2: &dyn Shape,
+    ) -> Result<bool, Unsupported> {
+        let dispatcher = DefaultQueryDispatcher;
+        dispatcher.intersection_test(pos12, g1, g2)
```

Comment:
call directly `intersection_test_ball_ball` to show custom dispatch ; only then fallbackto default's

Reply:
+ test if that workds :sweat_smile: